attestation-agent/attestation-agent/src/lib.rs

```diff
@@ -69,27 +76,43 @@ pub trait AttestationAPIs {
 
 /// Attestation agent to provide attestation service.
 pub struct AttestationAgent {
-    _config: Config,
+    config: Config,
+    attester: BoxedAttester,
+    eventlog: Mutex<EventLog>,
 }
 
-impl Default for AttestationAgent {
-    /// This function would panic if a malformed `aa_kbc_param` is given
-    /// either env or kernel cmdline
-    fn default() -> Self {
-        if let Ok(_config) = Config::try_from(config::DEFAULT_AA_CONFIG_PATH) {
-            return AttestationAgent { _config };
-        }
+impl AttestationAgent {
+    pub async fn init(&mut self) -> Result<()> {
+        // We should get the current platform's evidence to see the RTMR value.
+        // Here we assume RTMR is not polluted thus all be set `\0`
+        let init_entry = match self.config.eventlog_config.eventlog_algorithm {
+            HashAlgorithm::Sha256 => "INIT sha256/0000000000000000000000000000000000000000000000000000000000000000",
+            HashAlgorithm::Sha384 => "INIT sha384/000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
+            HashAlgorithm::Sha512 => "INIT sha512/00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
+        };
 
-        AttestationAgent {
-            _config: Config::new().expect("AA initialize"),
-        }
+        let event_digest = self
+            .config
+            .eventlog_config
+            .eventlog_algorithm
+            .digest(init_entry.as_bytes());
+
+        let tee_type = detect_tee_type();
+        let attester = TryInto::<BoxedAttester>::try_into(tee_type)?;
+        let mut eventlog = self.eventlog.lock().await;
+
+        attester
```

Comment:
we should be able to use `self.attester` here

Reply:
Yes, fixed